General/OOD/SemanticConstructorOverloading/SemanticConstructorOverloading.Problem/Optional Constructor Parameter Collision/Good/SampleClass.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+
+namespace SemanticConstructorOverloading.Problem.Optional_Constructor_Parameter_Collision.Good
+{
+    /// <summary>
+    /// Optional Constructor parameters cause problems when they are of the same data type.
+    /// This is a better implementation than the alternative, but it's still ugly, and 
+    /// not really recommended.
+    /// </summary>
+    public class SampleClass
+    {
+        // Public property assignments managed by constructor. In this use case, these will most
+        // likely be used for some kind of object configuration option.
+        public string StringValueA { get; set; }
+        public string StringValueB { get; set; }
+
+        /// <summary>
+        /// When two parameters of the same type are OPTIONAL in a constructor, specifying them as 
```

Comment:
> of the same type

Does it really matter if they are of the same type? Is the number two important? What about two or more?